R/census_vectors.R

```diff
@@ -0,0 +1,221 @@
+        grepl("^3.", add) ~ gsub(".", ", ", gsub("^3\\.", "Median of ", add),
+                                 fixed = TRUE),
+        grepl("^4.", add) ~ gsub(".", ", ", gsub("^4\\.", "Average to ", add),
+                                 fixed = TRUE)
+      )) %>%
+      dplyr::select(.data$vector, .data$type, .data$label, .data$units,
+                    parent_vector = .data$parent, .data$aggregation,
+                    .data$details)
+    attr(result, "last_updated") <- Sys.time()
+    attr(result, "dataset") <- dataset
+    save(result, file = cache_file)
+    result
+  } else {
+    if (!quiet) message("Reading vector information from local cache.")
+    load(file = cache_file)
+    last_updated <- attr(result, "last_updated")
+    if (!quiet && is.null(last_updated) ||
+        difftime(Sys.time(), last_updated, units = "days") > 1) {
+      warning(paste("Cached vectors list may be out of date. Set `use_cache =",
+                    "FALSE` to update it."))
+    }
+    attr(result, "dataset") <- dataset # just in case, catching cached legacy datasets
+    result
+  }
+}
+
+#' List all parent variables from vector hierarchical based on a (sub-)list of census
+#' variables returned by
+#' \code{list_census_vectors} or \code{search_census_vectors}.
+#'
+#' @param vector_list The list of vectors to be used
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' list_census_vectors("CA16") %>%
+#'   filter(vector == "v_CA16_4092") %>%
+#'   parent_census_vectors()
+parent_census_vectors <- function(vector_list){
+  base_list <- vector_list
+  dataset <- attr(base_list, "dataset")
+  n=0
+  vector_list <-
+    list_census_vectors(dataset, use_cache = TRUE, quiet = TRUE) %>%
+    dplyr::filter(vector %in% base_list$parent_vector) %>%
+    dplyr::distinct(vector, .keep_all = TRUE)
+  while (n!=nrow(vector_list)) {
+    n=nrow(vector_list)
+    new_list <- list_census_vectors(dataset, use_cache = TRUE, quiet = TRUE) %>%
+      dplyr::filter(vector %in% vector_list$parent_vector)
+    vector_list <- vector_list %>% rbind(new_list) %>%
+      dplyr::distinct(vector, .keep_all = TRUE)
+  }
+  attr(vector_list, "dataset") <- dataset
+  return(vector_list)
+}
+
+#' List all child variables from vector hierarchical based on a (sub-)list of census
+#' variables returned by
+#' \code{list_census_vectors} or \code{search_census_vectors}.
+#'
+#' @param vector_list The list of vectors to be used
+#' @param leaves_only Boolean flag to indicate if only leaf vectors should be returned,
+#' i.e. vectors that don't have children
+#' @param max_level optional, maximum depth to look for child vectors. Default is NA will return all
+#' child census vectors..
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' list_census_vectors("CA16") %>%
+#'   filter(vector == "v_CA16_4092") %>%
+#'   child_census_vectors(TRUE)
+child_census_vectors <- function(vector_list, leaves_only=FALSE,max_level=NA){
```

Comment:
Tested both old and new method here, and works nicely. The examples need to be updated though with the new functionality, also that particular vector doesn't have children oops